sdks/java/io/jms/src/test/java/org/apache/beam/sdk/io/jms/JmsIOIT.java

```diff
@@ -232,12 +243,13 @@ private PipelineResult readMessages() {
     pipelineRead
         .apply(
             "Read Messages",
-            JmsIO.<String>readMessage()
+            commonJms
```

Comment:
our integration test is also served as example that how a specific IO can be used. Could you please either keep `.withConnectionFactory(connectionFactory)` or just use withConnectionFactoryFn(...).

Currently it sets "withConnectionFactory" but serving a connectionFactoryProviderFn though its get method, sounds a little bit unnatural.

The point is the configuration of JmsIO.<String>readMessage() can be served as an example thus minimize test-taste settings.